core/kernel/notif_default.c

```diff
@@ -0,0 +1,98 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2021-2023, Linaro Limited
+ */
+
+#include <bitstring.h>
+#include <drivers/gic.h>
```

Comment:
replace with `#include <kernel/interrupt.h>`
add inclusion of _assert.h_ and _trace.h_.